httpbakery/client.go

```diff
@@ -313,6 +313,9 @@ func (c *Client) HandleError(reqURL *url.URL, err error) error {
 		return errgo.Notef(err, "cannot make cookie")
 	}
 	cookie.Path = cookiePath
+	if name := respErr.Info.CookieName; name != "" {
+		cookie.Name = "macaroon-" + name
```

Comment:
Because the code for checking macaroon cookies checks only the cookies with a "macaroon-" prefix, and we want this code to work against current APIs. If we didn't prefix the name here, then servers that didn't provide a name with the prefix wouldn't work.